extension/src/NABfunctions.ts

```diff
@@ -1536,3 +1536,74 @@ export async function renumberALObjects(): Promise<void> {
     showErrorAndLog("Renumber AL objects", error as Error);
   }
 }
+export async function troubleshootFindTransUnitsWithoutSource(): Promise<void> {
+  logger.log("Running: troubleshootFindTransUnitsWithoutSource");
+  Telemetry.trackEvent("troubleshootFindTransUnitsWithoutSource");
+
+  const settings = SettingsLoader.getSettings();
+  const appManifest = SettingsLoader.getAppManifest();
+  const gXlfFilePath = WorkspaceFunctions.getGXlfFilePath(
+    settings,
+    appManifest
+  );
+  const alObjects = await WorkspaceFunctions.getAlObjectsFromCurrentWorkspace(
+    settings,
+    appManifest,
+    true
+  );
+  const gXlf = Xliff.fromFileSync(gXlfFilePath);
+  const failingTransUnits: TransUnit[] = [];
+  gXlf.transunit.forEach((tu) => {
+    try {
+      const obj = DocumentFunctions.getObjectFromTokens(
+        alObjects,
+        tu.getXliffIdTokenArray()
+      );
+      const mlObjects = obj.getAllMultiLanguageObjects({
+        onlyForTranslation: true,
+      });
+      const mlObject = mlObjects.find(
+        (x) => x.xliffId().toLowerCase() === tu.id.toLowerCase()
+      );
+      if (!mlObject) {
+        throw new Error("ML Object not found");
+      }
+      // console.log(`${tu.id} => ${mlObject.text}`);
+    } catch (error) {
+      console.log(`Missing source: "${tu.id}"`);
+      failingTransUnits.push(tu);
+    }
+  });
+  if (failingTransUnits.length === 0) {
+    vscode.window.showInformationMessage(
+      "All is OK, could find the source of all TransUnits."
+    );
+  } else {
+    logger.error("The following TransUnits could not identify it's code:");
+    failingTransUnits.forEach((tu) =>
+      logger.log(`${tu.id} - ${tu.xliffGeneratorNoteContent()}`)
+    );
+    const result = LanguageFunctions.findTransUnitId(
```

Comment:
Maybe there's a better name than `result` for this name? Like first `firstFailing`, `foundFirstFailing` or something?

Reply:
Mainly because I read this bottom to top and had a hard time making sense of the last if/else block :sweat_smile:

Reply:
done